### PR TITLE
add js-dom project

### PR DESCRIPTION
**Pull request by JuliaNovak**
### js-dom

[Demo](https://julianovak.github.io/js-dom/) |
[Code base](https://github.com/JuliaNovak/js-dom)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by JuliaNovak** [comment]
have completed self-checks and fixed code accordingly

File: submissions/JuliaNovak/js-dom/script.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    },
+    {
+        id: 6,
+        type: 'Array.prototype.toString()',
+        description: 'The toString() method returns a string representing the specified array and its elements.',
+        img: './img/toString.png'
+    },
+]
+
+const methodsList = document.getElementsByClassName('methods-list')[0]
```
Comment (madmaxWMFU) [inline submissions/JuliaNovak/js-dom/script.js:40]: Use `document.querySelector()`

File: submissions/JuliaNovak/js-dom/script.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+methods.forEach(el => {
+    const li = document.createElement('li')
```
Comment (madmaxWMFU) [inline submissions/JuliaNovak/js-dom/script.js:42]: Read about [dataset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset)
Add the id value to the dataset property of the element.

File: submissions/JuliaNovak/js-dom/script.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    const title = document.createElement('h4')
+    title.innerText = el.type
+    title.classList.add("title");
+    
+    const description = document.createElement('p')
+    description.innerText = el.description
+
+    const methodsDescription = document.querySelector('.methods-description')
+    
+    li.addEventListener('click', () => {
```
Comment (madmaxWMFU) [inline submissions/JuliaNovak/js-dom/script.js:59]: Read about [event delegation](https://javascript.info/event-delegation).
After clicking, we get the id value of the dataset target element. Then we can use that id to find data from the array and render it.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.

File: submissions/JuliaNovak/js-dom/script.js (hunk 2 above)
Comment (JuliaNovak) [reply submissions/JuliaNovak/js-dom/script.js:42]: could you please explain the benefit of implementing the dataset property over other properties like classList?

Comment (madmaxWMFU) [reply submissions/JuliaNovak/js-dom/script.js:42]: Each of our links is unique, as is our dataset. So we can add a unique identifier to the dataset of the `li` element. Using the `addEventListener` on click that we assign to the menu, we define the selected item. We get the ID from its dataset. Then a corresponding element of the array is searched. We show the result to the user.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Divide your functionality into separate functions
```javascript
const menu = document.querySelector()

function renderMenuList() {
    // render li
}
function renderContent(data) {
   // data is a filter list by selected id in function onClick
   //  render content
}
function onClick() {
     //get target id and render content
}

function init() {
  renderMenuList()
  menu.addEventListener('click', onClick);
}

document.addEventListener("DOMContentLoaded", init);
```

File: submissions/JuliaNovak/js-dom/script.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+   description.innerText = data.description;
+
+   methodsDescription.appendChild(title);
+   methodsDescription.appendChild(description);
+   methodsDescription.appendChild(img);
+}
+
+function onClick({ target }) {
+   const selectedLi = target.dataset.id;
+   const el = methods.find((method) => method.id == selectedLi);
```
Comment (madmaxWMFU) [inline submissions/JuliaNovak/js-dom/script.js:77]: Do not forget about the informative naming of variables.
`const selectedMethod = .... ;`

**Pull request merged**